lib/msal-common/docs/performance.md

```diff
@@ -0,0 +1,71 @@
+# Performance
+
+This document will outline techniques your application can use to improve the performance of token acquisition using MSAL.
+
+## Bypass cloud instance discovery resolution
+
+> :information_source: This section does not apply if you are using **B2C**, **ADFS** or **dSTS** authorities. You will need to provide your authority domains to the `auth.knownAuthorities` property in the `PublicClientApplication`/`ConfidentialClientApplication` configuration object instead.
+>
+> * [msal-browser configration](../../msal-browser/docs/configuration.md#auth-config-options)
+> * [msal-node configuration](../../msal-browser/docs/configuration.md#auth-config-options)
+
+By default, during the process of retrieving a token, MSAL will make a network request to retrieve metadata associated with the various Azure clouds. If you would like to skip this network request, you can provide the required metadata in the configuration of `PublicClientApplication` or `ConfidentialClientApplication`.
+
+> :warning: It is your application's responsibility to ensure it is using correct, up-to-date cloud instance metadata. Failure to do so may result in your application not working correctly.
+
+### Instructions
+
+Instance Discovery Endpoint: `https://login.microsoftonline.com/common/discovery/instance?api-version=1.1&authorization_endpoint=https://login.microsoftonline.com/common/oauth2/v2.0/authorize`
+
+1. Make a request to the instance discovery endpoint above
+2. Provide the **entire** JSON response to the `auth.cloudDiscoveryMetadata` property.
+
+**Note** If none of the aliases listed in the response match your authority you should pass your authority domain in `auth.knownAuthorities` instead.
+
+Example:
+
+```js
+const msalInstance = new msal.PublicClientApplication({
```

Comment:
We only use the msal namespace in the CDN build. Let's leave it off to avoid confusion

```suggestion
const msalInstance = new PublicClientApplication({
```